### PR TITLE
Allow indexes to use inherited properties

### PR DESCRIPTION
**Pull request by smitpatel**
part of #2514

### CODE REVIEW

File: src/EntityFramework.Core/Metadata/Internal/EntityType.cs [hunk 1]
```diff
@@ -735,10 +735,10 @@ public virtual Index AddIndex([NotNull] IReadOnlyList<Property> properties)
             var duplicateIndex = FindIndexesInHierarchy(properties).FirstOrDefault();
             if (duplicateIndex != null)
             {
-                throw new InvalidOperationException(CoreStrings.DuplicateIndex(Property.Format(properties), DisplayName(), duplicateIndex.DeclaringEntityType.DisplayName()));
+                throw new InvalidOperationException(CoreStrings.DuplicateIndex(Property.Format(properties), Name, duplicateIndex.DeclaringEntityType.Name));
```
Comment (AndriySvyryd) [inline src/EntityFramework.Core/Metadata/Internal/EntityType.cs]: Why are you changing this?

File: src/EntityFramework.Core/Metadata/Internal/MetadataHelper.cs [hunk 2]
```diff
@@ -26,5 +27,18 @@ public static void CheckSameEntityType(IReadOnlyList<Property> properties, strin
                 }
             }
         }
+
+        public static void CheckPropertiesInEntityType(IReadOnlyList<Property> properties, EntityType entityType, string argumentName)
+        {
+            foreach (var property in properties)
+            {
+                if (property.DeclaringEntityType != entityType
+                    && entityType.FindProperty(property.Name) != property)
```
Comment (AndriySvyryd) [inline src/EntityFramework.Core/Metadata/Internal/MetadataHelper.cs]: You can make this check faster:
```
if(!property.DeclaringEntityType.IsAssignableFrom(entityType))
```

File: test/EntityFramework.Core.Tests/Metadata/Internal/IndexTest.cs [hunk 3]
```diff
@@ -41,14 +42,21 @@ public void Can_create_unique_index_from_properties()
         [Fact]
         public void Constructor_check_arguments()
         {
+            // ReSharper disable AssignNullToNotNullAttribute
```
Comment (AndriySvyryd) [inline test/EntityFramework.Core.Tests/Metadata/Internal/IndexTest.cs]: This test can be removed, we don't test argument validation anymore

File: src/EntityFramework.Core/Metadata/Internal/EntityType.cs (hunk 1 above)
Comment (smitpatel) [inline src/EntityFramework.Core/Metadata/Internal/EntityType.cs]: I will revert this for current PR. Filed #3583

**Comment by AndriySvyryd** [comment]
:shipit:

**Pull request merged**